components/motor/i2cmotors/ezopmp.go

```diff
@@ -292,8 +292,19 @@ func (m *Ezopmp) GoTo(ctx context.Context, mLPerMin, mins float64, extra map[str
 }
 
 // SetRPM instructs the motor to move at the specified RPM indefinitely.
-func (m *Ezopmp) SetRPM(ctx context.Context, rpm float64, extra map[string]interface{}) error {
-	return motor.NewSetRPMUnsupportedError(m.Name().ShortName())
+func (m *Ezopmp) SetRPM(ctx context.Context, mLPerMin float64, extra map[string]interface{}) error {
+	switch speed := math.Abs(mLPerMin); {
+	case speed < 0.1:
+		m.logger.CWarn(ctx, "motor speed is nearly 0 rev_per_min")
+		return motor.NewZeroRPMError()
+	case m.maxFlowRate > 0 && speed > m.maxFlowRate-0.1:
+		m.logger.CWarnf(ctx, "motor speed is nearly the max rev_per_min (%f)", m.maxFlowRate)
+	default:
+	}
+
+	powerPct := mLPerMin / m.maxFlowRate
```

Comment:
will this always be validated as non-zero and 105 ml/min from this driver's validate or constructor functions? if not, we should do that and set the motor's default.

Reply:
doesn't seem like there's a check for if it's over 105 but I can add one

Reply:
yeah let's do that, we might as well!

Reply:
well, there's a check for if mLPerMin is over the maxFlowRate (105) in the checks above, but not if the maxFlowRate is above 105